### PR TITLE
Get rid of util.StringList

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://github.com/openshift/origin/issues/4172

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
CC @deads2k

File: pkg/cmd/cli/secrets/add_secret_to_obj.go [hunk 1]
```diff
@@ -82,13 +80,7 @@ func NewCmdAddSecret(name, fullName string, f *cmdutil.Factory, out io.Writer) *
...
 	}
 
-	forFlag := &pflag.Flag{
-		Name:     "for",
-		Usage:    "type of secret to add: mount or pull",
-		Value:    &typeFlags,
-		DefValue: "mount",
-	}
-	cmd.Flags().AddFlag(forFlag)
+	cmd.Flags().StringSliceVar(&typeFlags, "for", []string{"mount"}, "type of secret to add: mount or pull")
```
Comment (deads2k) [inline pkg/cmd/cli/secrets/add_secret_to_obj.go]: reminder for me to come back and look at this line.

File: pkg/cmd/server/start/command_test.go [hunk 2]
```diff
@@ -170,9 +171,9 @@ func TestCommandBindingNodesForMaster(t *testing.T) {
 	actualCfg := executeMasterCommand([]string{"master", "--nodes=" + valueToSet})
 
 	expectedArgs := NewDefaultMasterArgs()
-	expectedArgs.NodeList.Set(valueToSet)
+	expectedArgs.NodeList = append(expectedArgs.NodeList, strings.Split(valueToSet, ",")...)
 
-	if expectedArgs.NodeList.String() != actualCfg.NodeList.String() {
+	if fmt.Sprint(expectedArgs.NodeList) != fmt.Sprint(actualCfg.NodeList) {
```
Comment (deads2k) [inline pkg/cmd/server/start/command_test.go]: I'd ask who wrote it, but I'm afraid of the answer.  Since you're here, can you fix it?

**Comment by deads2k** [comment]
Looks pretty straight forward.  There's one line I want to go back to, but I'll do that after the 3.1 push.

**Comment by fabianofranz** [comment]
@deads2k bump

**Comment by deads2k** [comment]
retitle the first commit `UPSTREAM: revert: 199adb7: <drop>: add back flag types to reduce noise during`, and merge at will

**Comment by fabianofranz** [comment]
[merge]

**Pull request merged**